From ea953b501219fcfff945fc36383012f427f5c771 Mon Sep 17 00:00:00 2001
From: Dmitry Shachnev <mitya57@gmail.com>
Date: Sat, 27 Jun 2020 17:03:35 +0300
Subject: [PATCH] Fix build on MIPS

On MIPS systems, "mips" is a built-in compiler macro. Undefine it to fix
compilation errors:

app/perfregisterinfo.cpp:55:12: error: expected unqualified-id before numeric constant
   55 | static int mips[] = { 32,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
      |            ^~~~

Change-Id: I80680337a019629e53828d60fc8a233a7cc819d2
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
---
 app/perfregisterinfo.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/3rdparty/perfparser/app/perfregisterinfo.cpp b/3rdparty/perfparser/app/perfregisterinfo.cpp
index 2e69d5e..bc0831d 100644
--- a/3rdparty/perfparser/app/perfregisterinfo.cpp
+++ b/3rdparty/perfparser/app/perfregisterinfo.cpp
@@ -52,6 +52,10 @@ static int aarch64[] = { 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13,
 static int x86[] = {0, 2, 3, 1, 7, 6, 4, 5, 8};
 static int x86_64[] = {0, 3, 2, 1, 4, 5, 6, 7, 16, 17, 18, 19, 20, 21, 22, 23, 8};
 
+#ifdef mips
+// On MIPS systems, "mips" is a built-in compiler macro.
+#undef mips
+#endif
 static int mips[] = { 32,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
                         18, 19, 20, 21, 22, 23, 24, 25, 28, 29, 30, 31};
 
-- 
2.30.2
